src/Tasks.UnitTests/RoslynCodeTaskFactory_Tests.cs

```diff
@@ -19,6 +27,102 @@ public class RoslynCodeTaskFactory_Tests
     {
         private const string TaskName = "MyInlineTask";
 
+        [Fact]
+        public void InlineTaskWithAssemblyPlatformAgnostic()
+        {
+            using (TestEnvironment env = TestEnvironment.Create())
+            {
+                TransientTestFolder folder = env.CreateFolder(createFolder: true);
+                string location = Assembly.GetExecutingAssembly().Location;
+                TransientTestFile inlineTask = env.CreateFile(folder, "5106.proj", @$"
+<Project>
+
+  <UsingTask TaskName=""MyInlineTask"" TaskFactory=""RoslynCodeTaskFactory"" AssemblyFile=""$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll"">
+    <Task>
+      <Reference Include=""{ Path.Combine(Path.GetDirectoryName(location), "..", "..", "..", "Samples", "Dependency", location.Contains("Debug") ? "Debug" : "Release", "net472", "Dependency.dll") }"" />
+      <Using Namespace=""Dependency"" />
+      <Code Type=""Fragment"" Language=""cs"" >
+<![CDATA[
+Log.LogError(Alpha.GetString());
+]]>
+      </Code>
+    </Task>
+  </UsingTask>
+
+<Target Name=""ToRun"">
+  <MyInlineTask/>
+</Target>
+
+</Project>
+");
+                string output = RunnerUtilities.ExecMSBuild(inlineTask.Path, out bool success);
+                success.ShouldBeTrue(output);
+                output.ShouldContain("Alpha.GetString");
+            }
+        }
+
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.AnyUnix, ".NETFramework 4.0 isn't on unix machines.")]
+        public void InlineTaskWithAssembly()
```

Comment:
Does this add anything over the cross-plat version now?

Reply:
I regard this method of building an assembly during tests as unreliable and suspect, because it's so different from "normal" projects. I don't understand the "start loading Dependency" comments--we're starting a fresh process for the test so how would it have loaded Dependency?

Reply:
If it turns out to be a flaky test, we can remove it. Good enough?

Reply:
It's a fresh process, but it's a fresh MSBuild process, so it automatically knows about things next to itself. If we start including Dependency.dll next to MSBuild for some reason.